tools/src/test/java/org/apache/kafka/tools/consumer/group/ConsumerGroupExecutor.java

```diff
@@ -0,0 +1,211 @@
+                syncCommit
+        );
+    }
+
+    public static ConsumerGroupExecutor buildClassicGroup(String brokerAddress,
+                                                          int numConsumers,
+                                                          String groupId,
+                                                          String topic,
+                                                          String assignmentStrategy,
+                                                          Optional<Properties> customConfigs,
+                                                          boolean syncCommit) {
+        return new ConsumerGroupExecutor(
+                brokerAddress,
+                numConsumers,
+                groupId,
+                GroupProtocol.CLASSIC.name,
+                topic,
+                assignmentStrategy,
+                Optional.empty(),
+                customConfigs,
+                syncCommit
+        );
+    }
+
+    void submit(ConsumerRunnable consumerThread) {
+        consumers.add(consumerThread);
+        executor.execute(consumerThread);
+    }
+
+    @Override
+    public void close() throws Exception {
+        consumers.forEach(ConsumerRunnable::shutdown);
+        executor.shutdown();
+        try {
+            executor.awaitTermination(5000, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static class ConsumerRunnable implements Runnable {
+        private final String brokerAddress;
+        private final String groupId;
+        private final Properties customConfigs;
+        private final boolean syncCommit;
+        private final String topic;
+        private final String groupProtocol;
+        private final String assignmentStrategy;
+        private final Optional<String> remoteAssignor;
+        private final Properties props = new Properties();
+        private KafkaConsumer<String, String> consumer;
+        private boolean configured = false;
+        private volatile boolean isShutdown = false;
+
+        public ConsumerRunnable(String brokerAddress,
+                                String groupId,
+                                String groupProtocol,
+                                String topic,
+                                String assignmentStrategy,
+                                Optional<String> remoteAssignor,
+                                Optional<Properties> customConfigs,
+                                boolean syncCommit) {
+            this.brokerAddress = brokerAddress;
+            this.groupId = groupId;
+            this.customConfigs = customConfigs.orElse(new Properties());
+            this.syncCommit = syncCommit;
+            this.topic = topic;
+            this.groupProtocol = groupProtocol;
+            this.assignmentStrategy = assignmentStrategy;
+            this.remoteAssignor = remoteAssignor;
+
+            this.configure();
```

Comment:
We don't need this method, since it is used only once. For example:
```java
            Map<String, Object> configs = new HashMap<>();
            configs.put(BOOTSTRAP_SERVERS_CONFIG, brokerAddress);
            configs.put(GROUP_ID_CONFIG, groupId);
            configs.put(KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
            configs.put(VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
            configs.put(GROUP_PROTOCOL_CONFIG, groupProtocol);
            if (Objects.equals(groupProtocol, CONSUMER.toString())) {
                remoteAssignor.ifPresent(assignor -> configs.put(GROUP_REMOTE_ASSIGNOR_CONFIG, assignor));
            } else {
                configs.put(PARTITION_ASSIGNMENT_STRATEGY_CONFIG, assignmentStrategy);
            }
            configs.putAll(customConfigs);
            consumer = new KafkaConsumer<>(configs);
```